### ISSUE TITLE
get_raw_jwt returning None

### ISSUE THREAD

**Issue by sarangdharmapurikar1** [vimalloc/flask-jwt-extended#316]
Hi! 
I have a simple function

```
@app.route('/logmein', methods=['GET', 'POST'])
#@jwt_required
def logmein():  
  if verbose:
       app.logger.info("Inside logmein")
        token = request.headers['Authorization']
        app.logger.info("Token with bearer: " + token)
        raw_jwt = get_raw_jwt()
        app.logger.info("Raw JWT: " + json.dumps(raw_jwt, indent=4))
        verify_jwt_in_request()
        verify_fresh_jwt_in_request()
```

But even when I have a valid token, the get_raw_jwt is returning {}. I verified the token is valid by going to jwt.io and cut-pasting my token to decode it. It decodes fine. Here's the output of my def:

```
NFO:flask.app:Inside logmein
INFO:flask.app:Token with bearer: Bearer eyJ...<removed the remaining token for privacy>
INFO:flask.app:Raw JWT: {}
```

How can I debug this? Also,  verify_jwt_in_request() etc return no exceptions or anything? 
I have PROPAGATE_EXCEPTIONS = True in my app config.

Appreciate help!

**Comment by vimalloc** [comment]
`verify_jwt_in_request()` needs to be run before `get_raw_jwt` in order for the token to be parsed and saved for this request. I’m looking at changing the `get_raw_jwt` behavior in the upcoming 4.0.0 to raise an error if the token does not exist, in order to make these situations more clear :+1:

**Comment by vimalloc** [comment]
The aforementioned change has been merged into the 4.0.0-dev branch, and will be going out with the next major release :+1:
